src/App.js

```diff
@@ -27,24 +27,7 @@ const App = () => {
     <BrowserRouter>
       <div>
         <NavBar query={query} updateQuery={updateQuery} getSearch={getSearch} />
-        <Routes>
-          <Route
-            path="/"
-            element={
-              <Home loading={loading} error={error} videoData={videoData} />
-            }
-          />
-          <Route
-            path="video-details/:id"
-            element={
-              <VideoDetails
-                loading={loading}
-                error={error}
-                videoData={videoData}
-              />
-            }
-          />
-        </Routes>
+        <AppRouter loading={loading} error={error} videoData={data} />
```

Comment:
I think the best would be to not pass this props to the router, instead do the fetch inside Home